arch/arm/soc/nordic_nrf/nrf52/soc.c

```diff
@@ -25,354 +25,16 @@ extern void _NmiInit(void);
-{
-	if ((*(u32_t *)0x10000130ul == 0x8ul) &&
-	    (*(u32_t *)0x10000134ul == 0x0ul)) {
-		return true;
-	}
-
-	return false;
-}
-
-static bool errata_103(void)
-{
-	if ((*(u32_t *)0x10000130ul == 0x8ul) &&
-	    (*(u32_t *)0x10000134ul == 0x0ul)) {
-		return true;
-	}
-
-	return false;
-}
-
-static bool errata_115(void)
-{
-	if ((*(u32_t *)0x10000130ul == 0x8ul) &&
-	    (*(u32_t *)0x10000134ul == 0x0ul)) {
-		return true;
-	}
-
-	return false;
-}
-
-static bool errata_120(void)
-{
-	if ((*(u32_t *)0x10000130ul == 0x8ul) &&
-	    (*(u32_t *)0x10000134ul == 0x0ul)) {
-		return true;
-	}
-
-	return false;
-}
-
-static bool errata_136_nrf52840(void)
-{
-	if ((*(u32_t *)0x10000130ul == 0x8ul) &&
-	    (*(u32_t *)0x10000134ul == 0x0ul)) {
-		return true;
-	}
-
-	return false;
-}
-
-static void nordicsemi_nrf52840_init(void)
-{
-	/* Workaround for Errata 36 "CLOCK: Some registers are not reset when
-	 * expected" found at the Errata document for your device located at
-	 * https://infocenter.nordicsemi.com/
-	 */
-	if (errata_36()) {
-		NRF_CLOCK->EVENTS_DONE = 0;
-		NRF_CLOCK->EVENTS_CTTO = 0;
-		NRF_CLOCK->CTIV = 0;
-	}
-
-	/* Workaround for Errata 98 "NFCT: Not able to communicate with the
-	 * peer" found at the Errata document for your device located at
-	 * https://infocenter.nordicsemi.com/
-	 */
-	if (errata_98()) {
-		*(volatile u32_t *)0x4000568Cul = 0x00038148ul;
-	}
-
-	/* Workaround for Errata 103 "CCM: Wrong reset value of CCM
-	 * MAXPACKETSIZE" found at the Errata document for your device
-	 * located at https://infocenter.nordicsemi.com/
-	 */
-	if (errata_103()) {
-		NRF_CCM->MAXPACKETSIZE = 0xFBul;
-	}
-
-	/* Workaround for Errata 115 "RAM: RAM content cannot be trusted upon
-	 * waking up from System ON Idle or System OFF mode" found at the
-	 * Errata document for your device located at
-	 * https://infocenter.nordicsemi.com/
-	 */
-	if (errata_115()) {
-		*(volatile u32_t *)0x40000EE4 =
-			(*(volatile u32_t *) 0x40000EE4 & 0xFFFFFFF0) |
-			(*(u32_t *)0x10000258 & 0x0000000F);
-	}
-
-	/* Workaround for Errata 120 "QSPI: Data read or written is corrupted"
-	 * found at the Errata document for your device located at
-	 * https://infocenter.nordicsemi.com/
-	 */
-	if (errata_120()) {
-		*(volatile u32_t *)0x40029640ul = 0x200ul;
-	}
-
-	/* Workaround for Errata 136 "System: Bits in RESETREAS are set when
-	 * they should not be" found at the Errata document for your device
-	 * located at https://infocenter.nordicsemi.com/
-	 */
-	if (errata_136_nrf52840()) {
-		if (NRF_POWER->RESETREAS & POWER_RESETREAS_RESETPIN_Msk) {
-			NRF_POWER->RESETREAS = ~POWER_RESETREAS_RESETPIN_Msk;
-		}
-	}
-
-	/* Configure GPIO pads as pPin Reset pin if Pin Reset capabilities
-	 * desired.
-	 * If CONFIG_GPIO_AS_PINRESET is not defined, pin reset will not be
-	 * available. One GPIO (see Product Specification to see which one) will
-	 * then be reserved for PinReset and not available as normal GPIO.
-	 */
-#if defined(CONFIG_GPIO_AS_PINRESET)
```

Comment:
Don't we need this to enable pin reset?

Reply:
We do. And this code is in `SystemInit()`: [here](https://github.com/zephyrproject-rtos/zephyr/blob/master/ext/hal/nordic/nrfx/mdk/system_nrf52.c#L199) for nRF52832 and [here](https://github.com/zephyrproject-rtos/zephyr/blob/master/ext/hal/nordic/nrfx/mdk/system_nrf52840.c#L168) for nRF52840.

Reply:
Got it, thanks!